### DISCUSSION TITLE
Cluster-wide access false not working as intended?

### DISCUSSION THREAD

**Discussion by m8-t** [kiali/kiali#7804] [Q&A]
Hi!

I just deployed kiali onto one of our clusters with lots of workloads. After a lot of testing and configuring I decided to limit kiali in terms of accessing namespaces.

So I set a discoverySelector in istios meshConfig as suggested in the documentation and configured the kiali CR like this:

```
    accessible_namespaces:
    - istio-system
    - prj-.*
    cluster_wide_access: false
```

I then deployed the bookinfo application and enabled istio on that namespace and it showed up as expected in kiali UI.

What I don't understand is why does kiali deploy a role in any namespace of the cluster? Right now I do not have any enabled istio namespaces but the kiali operator deployed a role in any namespace that matches `accessible_namespaces` but none of these namespaces match the discoverySelector of istio right now because nothing is enabled nor has the label the discoverySelector is looking for.

Is there a reason for this or did I misunderstand something RBAC wise?

thanks

**Comment by jshaughn** [comment]
Kiali does not "inherit" the Discovery Selectors defined for Istio.  You will need to configure Kiali's Discovery Selectors separately.

**Reply by jmazzitelli** [reply]
> Kiali does not "inherit" the Discovery Selectors defined for Istio. You will need to configure Kiali's Discovery Selectors separately.

Kiali doesn't have discovery selectors (yet) - that's coming in v2.0.

What @jshaughn meant to say was:

Kiali does not "inherit" the Discovery Selectors defined for Istio. Kiali's accessible_namespaces operates independently. They are separate things. In other words, Kiali knows nothing about the Istio discoverySelectors when setting up Kiali's roles.
